fission/src/mirabuf/MirabufLoader.ts

```diff
@@ -1,26 +1,125 @@
 import { mirabuf } from "../proto/mirabuf"
 import Pako from "pako"
-// import * as fs from "fs"
+
+const robots = "Robots"
+const fields = "Fields"
+const root = await navigator.storage.getDirectory()
+const robotFolderHandle = await root.getDirectoryHandle(robots, { create: true })
+const fieldFolderHandle = await root.getDirectoryHandle(fields, { create: true })
 
 export function UnzipMira(buff: Uint8Array): Uint8Array {
+    // Check if file is gzipped via magic gzip numbers 31 139
     if (buff[0] == 31 && buff[1] == 139) {
         return Pako.ungzip(buff)
     } else {
         return buff
     }
 }
 
-export async function LoadMirabufRemote(
-    fetchLocation: string,
-    useCache: boolean = true
-): Promise<mirabuf.Assembly | undefined> {
-    const miraBuff = await fetch(encodeURI(fetchLocation), useCache ? undefined : { cache: "no-store" })
-        .then(x => x.blob())
-        .then(x => x.arrayBuffer())
-    const byteBuffer = UnzipMira(new Uint8Array(miraBuff))
-    return mirabuf.Assembly.decode(byteBuffer)
+export async function LoadMirabufRemote(fetchLocation: string, type: MiraType): Promise<mirabuf.Assembly | undefined> {
+    const map = GetMap(type)
+    const targetID = map != undefined ? map[fetchLocation] : undefined
+
+    if (targetID != undefined) {
+        console.log("Loading mira from cache")
+        return (await LoadMirabufCache(fetchLocation, targetID, type, map)) ?? LoadAndCacheMira(fetchLocation, type)
+    } else {
+        console.log("Loading and caching new mira")
+        return await LoadAndCacheMira(fetchLocation, type)
+    }
 }
 
 // export function LoadMirabufLocal(fileLocation: string): mirabuf.Assembly {
 //     return mirabuf.Assembly.decode(UnzipMira(new Uint8Array(fs.readFileSync(fileLocation))))
 // }
+
+export async function ClearMira() {
+    for await (const key of robotFolderHandle.keys()) {
+        robotFolderHandle.removeEntry(key)
+    }
+    for await (const key of fieldFolderHandle.keys()) {
+        fieldFolderHandle.removeEntry(key)
+    }
+
+    window.localStorage.removeItem(robots)
+    window.localStorage.removeItem(fields)
+}
+
+export function GetMap(type: MiraType): any {
```

Comment:
Formatter getting upset at `any` here. Is this something we can change?

Reply:
I would create a type for the map. Something like `type MiraCache = { [path: string]: string }`, that way `GetMap` could return `MiraCache | undefined`. I also think `GetMap` could have a more descriptive name like `GetCache` because it's not immediately obvious what it does without reading the code.